bindings/rust/evmc-client/src/loader.rs

```diff
@@ -0,0 +1,74 @@
+/* EVMC: Ethereum Client-VM Connector API.
+ * Copyright 2019 The EVMC Authors.
+ * Licensed under the Apache License, Version 2.0.
+ */
+
+use evmc_sys as ffi;
+use libloading::{Library, Symbol};
+use std::ptr;
+use std::str;
+
+type EvmcCreate = extern "C" fn() -> *mut ffi::evmc_vm;
+
+#[derive(Debug)]
+pub enum EvmcLoaderErrorCode {
+    /** The loader succeeded. */
+    EvmcLoaderSucces = 0,
+
+    /** The loader cannot open the given file name. */
+    EvmcLoaderCannotOpen = 1,
+
+    /** The VM create function not found. */
+    EvmcLoaderSymbolNotFound = 2,
+
+    /** The invalid argument value provided. */
+    EvmcLoaderInvalidArgument = 3,
+
+    /** The creation of a VM instance has failed. */
+    EvmcLoaderInstanceCreationFailure = 4,
+
+    /** The ABI version of the VM instance has mismatched. */
+    EvmcLoaderAbiVersionMismatch = 5,
+
+    /** The VM option is invalid. */
+    EvmcLoaderInvalidOptionName = 6,
+
+    /** The VM option value is invalid. */
+    EvmcLoaderInvalidOptionValue = 7,
+}
+
+pub fn evmc_load_and_create(fname: &str) -> (*mut ffi::evmc_vm, EvmcLoaderErrorCode) {
```

Comment:
Why did you chose to reimplement this in Rust and not just use the C loader which already exists?

Reply:
I reimplement it is because I use this function (loader) outside evmc module at first. I want to avoid move partial evmc's C code to my codebase. But I agree it's better and more consistent to use the C loader after I move my evmc relative code into evmc-client now. I will try to replace it use the C loader, but maybe this won't finish too soon.

Reply:
Finish 5b2b4f2, btw I need cmake in ci environment.